packages/babel-plugin/src/utils/Cache.tsx

```diff
@@ -0,0 +1,106 @@
+import { hash } from '@compiled/utils';
+
+import { PluginOptions } from '../types';
+
+export interface CacheOptions extends PluginOptions {
+  cache?: boolean;
+  maxSize?: number;
+}
+
+const defaultOptions = {
+  cache: true,
+  maxSize: 500,
+};
+
+export class Cache {
+  _options: CacheOptions & typeof defaultOptions;
+  _cache: InstanceType<typeof Map>;
+
+  constructor(options?: CacheOptions) {
+    // TODO: Replace `this._instanceKey` with `this.#instanceKey` once we upgrade to typescript >= 3.8
+    this._options = { ...defaultOptions, ...options };
+
+    // TODO: add logic for reading and writing saved cached file to this._cache here
```

Comment:
nit: instead of us leaving TODOs everywhere could we place it in a github issue instead? just because things will probably change between now and then and it will end up being more noise in the short term.. esp. if theres some bugs to fix here in the short term

Reply:
https://github.com/atlassian-labs/compiled/issues/297